pyqtgraph/graphicsItems/PlotDataItem.py

```diff
@@ -254,59 +384,80 @@ def setAlpha(self, alpha, auto):
 
-    def setLogMode(self, xMode, yMode):
+    def setLogMode(self, xState, yState):
         """
-        To enable log scaling for y<0 and y>0, the following formula is used:
-        
-            scaled = sign(y) * log10(abs(y) + eps)
-
-        where eps is the smallest unit of y.dtype.
-        This allows for handling of 0. values, scaling of large values,
-        as well as the typical log scaling of values in the range -1 < x < 1.
-        Note that for values within this range, the signs are inverted.
+        When log mode is enabled for the respective axis by setting `xState` or 
+        `yState` to `True, a mapping according to ``mapped = np.log10( value )`` 
+        is applied to the data. For negative or zero values, this results in a 
+        NaN value.
         """
-        if self.opts['logMode'] == [xMode, yMode]:
+        if self.opts['logMode'] == [xState, yState]:
             return
-        self.opts['logMode'] = [xMode, yMode]
-        self.xDisp = self.yDisp = None
+        self.opts['logMode'] = [xState, yState]
+        self._datasetMapped  = None  # invalidata mapped data
+        self._datasetDisplay = None  # invalidate display data
         self.updateItems(styleUpdate=False)
         self.informViewBoundsChanged()
 
-
-    def setDerivativeMode(self, mode):
-        if self.opts['derivativeMode'] == mode:
+    def setDerivativeMode(self, state):
+        """
+        ``state = True`` enables derivative mode, where a mapping according to
+        ``y_mapped = dy / dx`` is applied, with `dx` and `dy` representing the 
+        differences between adjacent `x` and `y` values.
+        """
+        if self.opts['derivativeMode'] == state:
             return
-        self.opts['derivativeMode'] = mode
-        self.xDisp = self.yDisp = None
+        self.opts['derivativeMode'] = state
+        self._datasetMapped  = None  # invalidata mapped data
+        self._datasetDisplay = None  # invalidate display data
         self.updateItems(styleUpdate=False)
         self.informViewBoundsChanged()
 
-    def setPhasemapMode(self, mode):
-        if self.opts['phasemapMode'] == mode:
+    def setPhasemapMode(self, state):
+        """
+        ``state = True`` enables phase map mode, where a mapping 
+        according to ``x_mappped = y`` and ``y_mapped = dy / dx``
+        is applied, plotting the numerical derivative of the data over the 
+        original `y` values.
+        """
+        if self.opts['phasemapMode'] == state:
             return
-        self.opts['phasemapMode'] = mode
-        self.xDisp = self.yDisp = None
+        self.opts['phasemapMode'] = state
+        self._datasetMapped  = None  # invalidata mapped data
+        self._datasetDisplay = None  # invalidate display data
         self.updateItems(styleUpdate=False)
         self.informViewBoundsChanged()
 
-    def setPointMode(self, mode):
-        if self.opts['pointMode'] == mode:
+    def setPointMode(self, state):
```

Comment:
Any memory what this is supposed to do?
I could not find any actual function, but PlotItem seems to actually call it.

Reply:
Looks like it references a checkbox in PlotItem's ui template in the `plotConfigTemplate_<binding>`, something involving `pointsGroup`, supported values are None, True and False.

None `pointsGroup.isChecked()` and `autoPointsCheck.isChecked()`
True `pointsGroup.isChecked()` and _not_ `autoPointsCheck.isChecked()`
False _not_ `pointsGroup.isChecked()`

always hated those template files, not sure what uses that particular one.

Reply:
I'll try and take a deeper dive into the git history and figure out the source of those, and see if it's something left-behind  from some past cleanup effort.